src/plugins/console/public/application/hooks/use_data_init/use_data_init.test.tsx

```diff
@@ -0,0 +1,159 @@
+
+  interface MockObjectData {
+    createdAt: number;
+    updatedAt: number;
+    text: string;
+  }
+
+  const callMockFuncWithArg = (arg: MockObjectData[] | Function) => ({
+    ...mockContextValue,
+    services: {
+      ...mockContextValue.services,
+      objectStorageClient: {
+        text: {
+          findAll: jest.fn(() => arg),
+          create: jest.fn((data: any) => mockedObject),
+        },
+      },
+    },
+  });
+
+  beforeAll(() => {
+    setStateMock = jest.fn();
+    useStateMock = (state: any) => [state, setStateMock];
+    mockContextValue = serviceContextMock.create();
+    dispatch = jest.fn();
+    (useEditorActionContext as jest.Mock).mockReturnValue(dispatch);
+    (useServicesContext as jest.Mock).mockReturnValue({
+      ...mockContextValue,
+      services: {
+        ...mockContextValue.services,
+        objectStorageClient: {
+          text: {
+            findAll: jest.fn(() => []),
+            create: jest.fn((data: any) => mockedObject),
+          },
+        },
+      },
+    });
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('calls dispatch with new objects, if no texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    const { result } = renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(dispatch).toHaveBeenCalledWith({
+      type: 'setCurrentTextObject',
+      payload: mockedObject,
+    });
+  });
+
+  it('calls dispatch with new objects, if texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    const mockObj: MockObjectData = {
+      createdAt: 1643277939899,
+      updatedAt: 1643277939900,
+      text: 'Mocked data',
+    };
+
+    (useServicesContext as jest.Mock).mockReturnValue(callMockFuncWithArg([mockObj]));
+
+    const { result } = renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(dispatch).toHaveBeenCalledWith({
+      type: 'setCurrentTextObject',
+      payload: mockObj,
+    });
+  });
+
+  it('call retry function', async () => {
+    const setErrorMock = jest.fn();
+    const useErrorMock: UseStateMock = (state: any) => [state, setErrorMock];
+
+    const setDoneMock = jest.fn();
+    const useDoneMock: UseStateMock = (state: any) => [state, setDoneMock];
+
+    const setRetryTokenMock = jest.fn();
+    const useRetryTokenMock: UseStateMock = (state: any) => [state, setRetryTokenMock];
+
+    jest
+      .spyOn(React, 'useState')
+      .mockImplementationOnce(useErrorMock)
+      .mockImplementationOnce(useDoneMock)
+      .mockImplementationOnce(useRetryTokenMock);
+
+    const { result } = renderHook(() => useDataInit(), {});
+
+    result.current.retry();
+
+    expect(setErrorMock).toHaveBeenCalledWith(null);
+    expect(setDoneMock).toHaveBeenCalledWith(false);
+    expect(setRetryTokenMock).toHaveBeenCalledWith({});
+  });
+
+  it('call error function', async () => {
```

Comment:
test names should reflect the essence and help in understanding the code. Ideally, if you look at the tests, you can roughly understand what is happening in the file being tested.

e.g. `call error function` -> `should update the error state if case of getting an exception while loading data.`

Reply:
valid for all tests

Reply:
Fixed. Thanks for the comment